docker/docker_image.go

```diff
@@ -18,8 +18,8 @@ type Image struct {
 
 // NewDockerImage returns a new Image interface type after setting up
 // a client to the registry hosting the given image.
-func NewDockerImage(img, certPath string, tlsVerify bool) (types.Image, error) {
-	s, err := newDockerImageSource(img, certPath, tlsVerify)
+func NewDockerImage(img string, dc Client) (types.Image, error) {
+	s, err := newDockerImageSource(img, dc)
```

Comment:
(the last `i.src.c` suggestion is admittedly really ugly; still, there’s no pretending our tests are black-box anyway, and the simplicity is somewhat appealing.)